include/aspect/parameters.h

```diff
@@ -308,6 +308,9 @@ namespace aspect
 
     double                         nonlinear_tolerance;
     double                         nonlinear_switch_tolerance;
+    std::string                    use_Newton_stabilisation_preconditioner;
```

Comment:
I feel like it is time to give the Newton handler its own `declare_parameters` and `parse_parameters` functions. This way we do not clutter the main parameters with things that are specific to the Newton handler. The melt handler already does this, and probably you can copy the same approach (see parameters.cc:1728, core.cc:544, melt.cc:1367). Also as you mentioned these can be `enum`